packages/cli/src/lib/webpack/run-webpack.js

```diff
@@ -78,14 +78,14 @@ async function prodBuild(env) {
 }
 
 function runCompiler(compiler) {
-	return new Promise((res, rej) => {
+	return new Promise(res => {
 		compiler.run((err, stats) => {
-			showStats(stats, true);
-
-			if (err || (stats && stats.hasErrors())) {
-				rej(`${red('\n\nBuild failed! \n\n')} ${err || ''}`);
+			if (err) {
+				error(err, 1);
 			}
 
+			showStats(stats, true);
+
```

Comment:
`showStats()`, in production builds, will also immediately exit upon coming across any errors; as such, there's no reason to ever reject this promise as we've already exited.